python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -1,11 +1,56 @@
 """Utility functions for sampling."""
 
 from collections import defaultdict
-from typing import Dict, Tuple, Union
+from typing import Dict, List, Tuple, Union
 
 import torch
 
 
+def unique_and_compact_nodes_list(
+    nodes: Union[
+        List[torch.Tensor],
+        Dict[str, List[torch.Tensor]],
+    ],
+):
+    """
+    Compact a list of nodes tensor.
+
+    Parameters
+    ----------
+    nodes : List[torch.Tensor] or Dict[str, List[torch.Tensor]]
+        List of nodes for compacting.
+        - If `nodes` is a list of tensor: It means the graph is homogeneous.
```

Comment:
Usually the documentation should focus on how the method will handle the input instead of explain the meaning of the input. That says, the documentation here should focus on: if the input is a list of tensor, all the tensor will be unique_and_compact together while if the input is a dict of list, the unique_and_compact will be done per type. We can also mention "this is usually used for homogeneous graph / heterogeneous graph".